pkg/core/transaction/signer.go

```diff
@@ -15,11 +15,11 @@ const maxSubitems = 16
 
 // Signer implements a Transaction signer.
 type Signer struct {
-	Account          util.Uint160      `json:"account"`
-	Scopes           WitnessScope      `json:"scopes"`
-	AllowedContracts []util.Uint160    `json:"allowedcontracts,omitempty"`
-	AllowedGroups    []*keys.PublicKey `json:"allowedgroups,omitempty"`
-	Rules            []WitnessRule     `json:"rules,omitempty"`
+	Account          util.Uint160    `json:"account"`
+	Scopes           WitnessScope    `json:"scopes"`
+	AllowedContracts []util.Uint160  `json:"allowedcontracts,omitempty"`
+	AllowedGroups    keys.PublicKeys `json:"allowedgroups,omitempty"`
```

Comment:
Do you really need this change? It can break some initializations in existing code.

Although I've tried to search for uses of it and failed to find any... Old node code used it. Maybe @fyfyrchik can tell if it's still in use somewhere.

Reply:
okay, moved back to []*keys.PublicKey and adjusted Copy()